pkg/api/query/v1.go

```diff
@@ -394,12 +388,27 @@ func (qapi *QueryAPI) labelValues(r *http.Request) (interface{}, []error, *api.A
 		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("invalid label name: %q", name)}
 	}
 
+	start, err := parseTimeParam(r, "start", minTime)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+	}
+	end, err := parseTimeParam(r, "end", maxTime)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+	}
+	// Prometheus doesn't check this, do we need to?
```

Comment:
Sounds useful, no? (: 

Please ask comments on PR itself not in comment (: So we can still merge PR if code is ok.

Reply:
I think we should and it's probably worth contributing such a check upstream IMHO